File: js/src/samples/send_data/send_data.ts
import { Bloock, BloockClient, RecordBuilder } from "@bloock/sdk"
import { Config } from "../../utils/config"
import { Sample } from "../../utils/sample"
import { Logger } from "../../utils/logger";

Sample.run("Send Data", async (config: Config) => {
    // we set the API key and create a client
    Bloock.setApiKey(config.apiKey);
    let client = new BloockClient();

    // we create an array of strings which will contain
    // the hashes of the records we want to send
    let records = []
    let record = await RecordBuilder
        .fromString("Hello world")
        .build();
        
    // we can get the hash of the record
    let hash = await record.getHash();

    // and append it to the array
    records.push(hash);

    // finally we can send the records
    let sendReceipts = await client.sendRecords(records)

    // we get a receipt with informationa about the transaction
    Logger.success(`Record receipts: ${sendReceipts}`);
})

File: python/samples/send_data/send_data.py
import bloock
from bloock.client.client import Client
from bloock.client.builder import RecordBuilder

from utils.logger import Logger
from utils.sample import Sample
from utils.config import Config


def send_data(c: Config):
    bloock.api_key = c.api_key
    client = Client()

    record = RecordBuilder.from_string("Hello world").build()

    # we create an array of records which will contain the records we want to send
    records = [record]

    # we can get the hash of the record
    hash = record.get_hash()
    Logger.success("Hash: " + hash)

    # finally we can send the records
    send_receipts = client.send_records(records)

    # we get a receipt with informationa about the transaction
    Logger.success(f"Record receipts: {list(map(lambda x: x.__dict__, send_receipts))}")


Sample("send_data", send_data)
